doc/panes/editor.rst

```diff
@@ -29,32 +31,275 @@ The Editor pane consists of the following areas:
+
+
+Automatic formatting
+~~~~~~~~~~~~~~~~~~~~
+
+The Editor has built-in support for automatically formatting your code using several popular tools, including `Autopep8 <https://github.com/hhatto/autopep8>`_, `Yapf <https://github.com/google/yapf>`_, and `Black <https://black.readthedocs.io/en/stable/>`_.
+The :guilabel:`Format file or selection with {tool}` command in the :guilabel:`Source` or context menu will format either the selected fragment (if text is selected) or the entire active file.
+
+.. image:: /images/editor/editor-automatic-formatting.gif
+   :alt: Spyder editor panel, showing an example of code selection formatting
+
+You can have the Editor automatically autoformat a file every time you save your work.
+To set this up, go to :menuselection:`Preferences --> Completion and linting --> Code style and formatting --> Code formatting` and check the :guilabel:`Autoformat files on save` option.
+
+.. image:: /images/editor/editor-autoformat-setting.png
+   :alt: Spyder's preferences dialog, showing checking the autoformat files on save setting
+
+
+
+============
+Running code
+============
+
+The Editor lets you run an entire file as well as specific lines, selections or cells.
+
+As your code is running,
+
+* The :doc:`ipythonconsole` will display output and errors.
+* The :doc:`variableexplorer` allows you to browse and interact with the objects generated.
+* The :doc:`plots` pane renders the figures and images created.
+* The :doc:`profiler` helps you optimize your code's run time.
+
+
+Run file
+~~~~~~~~
+
+Run an entire Editor file using the :menuselection:`Run --> Run` menu item, the :guilabel:`Run file` toolbar button or the :kbd:`F5` key.
+Use :menuselection:`Run --> Re-Run last script` to re-run the most recent file executed with the above.
+
+
+Run line/selection
+~~~~~~~~~~~~~~~~~~
+
+You can execute the current line—or multiple selected lines—using the :guilabel:`Run selection or current line` option from the toolbar or the :menuselection:`Run` menu, as well as with the :kbd:`F9` key.
+After running the current line, the cursor automatically advances to the next, so you can step through your code line by line.
+Unlike :guilabel:`Run file`, the executed lines are shown in the :doc:`ipythonconsole`.
+
+
+Code cells
+~~~~~~~~~~
+
+To run a cell, place your cursor inside it and use the :menuselection:`Run --> Run cell` menu item, the :guilabel:`Run current cell` toolbar button or the :kbd:`Ctrl-Enter` / :kbd:`Cmd-Return` keyboard shortcut.
+Use :guilabel:`Run cell and advance` in the :guilabel:`Run` menu/toolbar or :kbd:`Shift-Enter` to jump to the next cell after running, useful for stepping through cells quickly.
+
+
+Run configuration
+~~~~~~~~~~~~~~~~~
+
+You can use the :guilabel:`Run configuration per file` dialog to set each file's working directory, console mode (current, dedicated or external), CLI arguments, execution options (clear all variables, run in an existing/empty namespace, debug on error), and more.
```

Comment:
I think that would be a great idea; IIRC this might have been mention in the original outline but I missed it here.

Here's a screenshot of the dialog showing the default settings for Spyder's default startup `temp.py` file.

![image](https://user-images.githubusercontent.com/17051931/201811605-abdc4ad1-eb04-45d2-87c5-fd7ffde0d1f0.png)

Reply:
@hyounes4560 I can take care of adding that, if you would like me to.

Reply:
I went ahead with adding this.